src/main/java/co/cask/hydrator/plugin/hive/action/HiveImport.java

```diff
@@ -43,24 +42,31 @@ public HiveImport(HiveConfig config) {
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    if (!config.containsMacro("statement")) {
+      validateHiveStatement(config.statement);
+    }
+  }
+
+  private void validateHiveStatement(String statement) {
     //validate hive command. For import we only accept Load statements
-    if (!config.statement.toUpperCase().startsWith("LOAD")) {
+    if (!statement.toUpperCase().startsWith("LOAD")) {
       throw new IllegalArgumentException("To import data to a hive table, please use Hive LOAD query");
     }
 
-    if (config.statement.length() < 15) {
+    if (statement.length() < 15) {
```

Comment:
the log messages could be a little more clear. like the error message for length, can say statement is longer than the allowed limit.